src/deck.js

```diff
@@ -31,6 +36,14 @@ class Deck {
   dealCards(player) {
     for (let i = player.cards.length; i < 6; i++) {
       player.cards.push(...this.cards.splice(0, 1));
+
+      if (this.$deck.childNodes.length > 0) {
+        if (player.name === 'Player1') {
+          this.$playerHand.appendChild(this.$deck.childNodes[0]);
+        } else {
+          this.$opponentHand.appendChild(this.$deck.childNodes[0]);
+        }
+      }
```

Comment:
Checking by name is bad idea, you can define player as first element of array, and check if it's index is 0